sdcm/nemesis.py

```diff
@@ -4318,6 +4319,131 @@ def disrupt_add_remove_mv(self):
+
+        During bootstrap new node stream data from token ring
+        If bootstrap is aborted, according to Failed topology
+        changes manual, the process could be restarted and node
+        should be added to cluster successfully. Nemesis terminate
+        the bootstrap streaming base on found log messages and then
+        remove ghost group0 members and restart bootstrap.
+
+        If node was not added anyway, clean it from cluster
+        and return the cluster to initial state(by num of nodes)
+        """
+        new_node: BaseNode = self.cluster.add_nodes(
+            count=1, dc_idx=self.target_node.dc_idx, enable_auto_bootstrap=True, rack=self.target_node.rack)[0]
+        self.monitoring_set.reconfigure_scylla_monitoring()
+        self.set_current_running_nemesis(node=new_node)  # prevent to run nemesis on new node when running in parallel
+
+        def start_bootstrap(new_node: BaseNode, timeout=3600):
+            bootstrap_process = Process(target=self.cluster.node_setup,
+                                        name=f"Bootstraping_{new_node.name}",
+                                        kwargs={"node": new_node, "verbose": True, "timeout": int(timeout)},
+                                        daemon=True)
+            try:
+                self.log.info("Start node %s setup and bootstrapp", new_node.name)
+                bootstrap_log_follower = new_node.follow_system_log(patterns=['init - Startup failed',
+                                                                              'init - Startup interrupted',
+                                                                              'initialization completed'])
+                bootstrap_process.start()
+                log_messages = wait_for(func=lambda: list(bootstrap_log_follower),
+                                        text='Waiting for bootstap aborting or finishing', step=5, timeout=timeout)
+
+                if log_messages and ('init - Startup failed' in log_messages[0] or 'init - Startup interrupted' in log_messages[0]):
+                    self.log.debug("Bootstrap node %s was aborted", new_node.name)
+                    raise NodeSetupFailed(node=new_node, error_msg=log_messages[0])
+
+                if new_node.db_up() and new_node.jmx_up():
+                    self.log.info("Node %s bootstrapped succesfull", new_node.name)
+            except Exception as exc:  # pylint: disable=broad-except
+                self.log.warning("Node bootstrap was aborted with error: %s", exc)
+            finally:
+                self.log.debug("Stop node bootstrap process")
+                bootstrap_process.join(60)
+                bootstrap_process.terminate()
+                self.log.debug("Node bootstrap process stopped")
+
+        trigger = partial(
+            start_bootstrap, new_node=new_node)
+
+        terminate_patterns = self.target_node.raft.get_log_pattern_for_bootstrap_abort()
+
+        self.use_nemesis_seed()
```

Comment:
If we always use nemesis_seed (at least per release) this test will always do the same thing, isn't it?

Reply:
Maybe we should split it to different nemesis, but the question if we're sure that we catch each of them in the correct place.

Reply:
or we cover all of them in the same nemesis, but we need a clear way to differ between them so when we report a failure of this nemesis we know which one exactly failed.

Reply:
> If we always use nemesis_seed (at least per release) this test will always do the same thing, isn't it?

yes, as for all other nemesis(like decommission_stream_error). But each job has its own nemesis_seed, hasn't it? the each call of new job will choose another log line. and if nemesis will be run several times in same job, it will choose different lines mainly

Reply:
Since I don't have a better suggestion at the moment, I think that the existing approach with the nemesis_seed is ok.
But if we want to exercise this nemesis several times in the same job, all we be done the same.

Reply:
Added